File: Insertion Sort/insertion_sort.c
// C program for insertion sort 
#include <math.h> 
#include <stdio.h> 

void insertionSort(int arr[], int num) 
{ 
	int i, k, j; 
	for (i = 1; i < num; i++) { 
		k = arr[i]; 
		j = i - 1; 

		while (j >= 0 && arr[j] > k) { 
			arr[j + 1] = arr[j]; 
			j = j - 1; 
		} 
		arr[j + 1] = k; 
	} 
} 

void printArray(int arr[], int num) 
{ 
	int i; 
	for (i = 0; i < num; i++) 
		printf("%d ", arr[i]); 
	printf("\n"); 
} 

int main() 
{ 
	int arr[] = { 21, 12, 23, 15, 5 }; 
	int num = sizeof(arr) / sizeof(arr[0]); 

	insertionSort(arr, num); 
	printArray(arr, num); 

	return 0; 
} 

File: Insertion Sort/insertion_sort.java
 class insertion_sort { 
	
	void sort(int arr[]) 
	{ 
		int num = arr.length; 
		for (int i = 1; i < num; ++i) { 
			int m = arr[i]; 
			int j = i - 1; 

			while (j >= 0 && arr[j] > m) { 
				arr[j + 1] = arr[j]; 
				j = j - 1; 
			} 
			arr[j + 1] = m; 
		} 
	} 

	static void printArray(int arr[]) 
	{ 
		int num = arr.length; 
		for (int i = 0; i < num; ++i) 
			System.out.print(arr[i] + " "); 
		System.out.println(); 
	} 

	public static void main(String args[]) 
	{ 
		int arr[] = {3,60,35,2,45,320,5}; 

		insertion_sort ob = new insertion_sort(); 
		ob.sort(arr); 

		printArray(arr); 
	} 
}
